docs/content/getting-started/releases.mdx

```diff
@@ -44,6 +44,16 @@ While technically the 0.y.z phase of Semantic Versioning is "anything goes", we
 
 ---
 
+# Python Version Support
+
+Each Dagster release strives to support the currently active versions of python.
+
+When a new version of python is released, Dagster will work to add support once Dagster's own core dependencies have been updated to support it.
+
+When a version of python reaches end of life, Dagster will drop support for it at the next convenient non-patch release.
+
+---
```

Comment:
This needs some more word-smithing but I believe this is roughly how we treat python version support. 

At one point i had something like "previously released versions of Dagster remain available for use with older python versions"  but pulled it.